pkg/diag/validator/pod.go

```diff
@@ -209,34 +209,35 @@ func processPodEvents(e corev1.EventInterface, pod v1.Pod, ps *podStatus) {
 	}
 	switch recentEvent.Reason {
 	case failedScheduling:
-		ps.statusCode = proto.StatusCode_STATUSCHECK_FAILED_SCHEDULING
-		ps.err = fmt.Errorf(recentEvent.Message)
+		ps.ae.ErrCode = proto.StatusCode_STATUSCHECK_FAILED_SCHEDULING
+		ps.ae.Message = recentEvent.Message
```

Comment:
These two-step struct accesses kinda go against the Law of Demeter — basically you should try to only manipulate information one step away.  But this function would make sense as a method on `podStatus *`. WDYT?